pkg/runtime/wfengine/client.go

```diff
@@ -164,23 +172,24 @@ func (c *client) Get(ctx context.Context, req *workflows.GetRequest) (*workflows
 	res := &workflows.StateResponse{
 		Workflow: &workflows.WorkflowState{
 			InstanceID:    req.InstanceID,
-			WorkflowName:  metadata.Name,
-			CreatedAt:     metadata.CreatedAt,
-			LastUpdatedAt: metadata.LastUpdatedAt,
-			RuntimeStatus: getStatusString(int32(metadata.RuntimeStatus)),
-			Properties: map[string]string{
-				"dapr.workflow.input":         metadata.SerializedInput,
-				"dapr.workflow.custom_status": metadata.SerializedCustomStatus,
-			},
+			WorkflowName:  metadata.GetName(),
+			CreatedAt:     metadata.GetCreatedAt().AsTime(),
+			LastUpdatedAt: metadata.GetLastUpdatedAt().AsTime(),
+			RuntimeStatus: getStatusString(int32(metadata.GetRuntimeStatus())),
 		},
 	}
 
+	if metadata.GetCustomStatus() != nil {
+		res.Workflow.Properties = map[string]string{
+			"dapr.workflow.custom_status": metadata.GetCustomStatus().GetValue(),
+		}
+	}
+
 	// Status-specific fields
+	//nolint:protogetter
 	if metadata.FailureDetails != nil {
-		res.Workflow.Properties["dapr.workflow.failure.error_type"] = metadata.FailureDetails.GetErrorType()
-		res.Workflow.Properties["dapr.workflow.failure.error_message"] = metadata.FailureDetails.GetErrorMessage()
-	} else if metadata.IsComplete() {
-		res.Workflow.Properties["dapr.workflow.output"] = metadata.SerializedOutput
+		res.Workflow.Properties["dapr.workflow.failure.error_type"] = metadata.GetFailureDetails().GetErrorType()
+		res.Workflow.Properties["dapr.workflow.failure.error_message"] = metadata.GetFailureDetails().GetErrorMessage()
```

Comment:
Potential `nil` panic if the condition is false in line 182 `if metadata.GetCustomStatus() != nil`, the `res.Workflow.Properties` won't be initialized.
Either guard both places that assign values with `nil` check of the map, or initialize the map in struct initialization above.